### DISCUSSION TITLE
How to set up Websocket for MeshCentral in Nginx Proxy Manager

### DISCUSSION THREAD

**Discussion by geekdomotique** [Ylianst/MeshCentral#4202] [Q&A]
Hello, (sorry for my poor english...)

I spend many hours to look for a solution with my websocket error when i try to login. 
I think the probleme is probably with my Nginx Proxy Manager(GUI) configuration, i'm not used to use it, i usually use my reverse proxy on my synology...
so, it would be great if someone could provide some screnshots to help me

Actually i have :

Details:
```
Domain name : mesh.xxx.fr
Scheme : http | Forward IP : 192.168.4.150 | Forward port : 4430
Cache assets, Blocks CE, Websockets support are activated
```

Custom Locations:
```
Location : /
Scheme : http | Forward IP : 192.168.4.150 | Forward port : 4430
in parameters i have add :
proxy_set_header Upgrade $http_upgrade;
proxy_set_header Connection "upgrade";
proxy_set_header Host $host;
proxy_set_header X-Forwarded-Host $host:$server_port;
proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
proxy_set_header X-Forwarded-Proto $scheme;
```

SSL :
```
Force SSL and HTTP/2 Support activated
```

My Config.json : 

```
{
  "$schema": "http://info.meshcentral.com/downloads/meshcentral-config-schema.json",
  "settings": {
    "cert": "mesh.xxx.fr",
    "WANonly": true,
    "_LANonly": true,
    "_sessionKey": "MyReallySecretPassword1",
    "port": 4430,
    "AliasPort": 443,
    "redirPort": 800,
    "_redirAliasPort": 80,
    "AgentPong": 300,
    "TLSOffload": "192.168.48.2", (My Nginx Proxy Manager(GUI) )
    "SelfUpdate": false,
    "AllowFraming": "false",
    "WebRTC": "false"
  },
  "domains": {
	"": {
	"_title": "MyServer",
    "_title2": "Servername",
    "_minify": true,
    "NewAccounts": "true",
	"_userNameIsEmail": true,
    "certUrl": "https://mesh.xxx.fr"
	}
  },
  "_letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before>",
    "_email": "myemail@mydomain.com",
    "_names": "myserver.mydomain.com",
	"production": false
  }
}
```

Thank for your support...

**Comment by geekdomotique** [comment]
Nobody use Nginx Proxy Manager?

**Comment by Ylianst** [comment]
I don't user NGINX Proxy Manager, but I did spot a video someone made that talks about [MeshCentral with NGINX Proxy manager here](https://www.youtube.com/watch?v=V0uDXCua4q8). I would imagine that once you understand the basic concepts of how MeshCentral works with reverse proxies, it should be possible to set it up with any reverse proxy. [I have a video on MeshCentral with NGINX here that goes thru the basics](https://www.youtube.com/watch?v=YSmiLyKSX2I).

- You need to forward external traffic to the main MeshCentral port.
- You need to configure MeshCentral to perform TLS or not and indicate the internal IP address of the reverse proxy so MeshCentreal knows to trust the "X-Forward-For" header.
- You also need to give MeshCentral a HTTPS URL it can use to get the external certificate that MeshAgents will see when connecting.

That last step is not typical for regular web applications, but MeshCentral needs this to prevent man-in-the-middle attacks on agent connections.

Hope this helps.

**Reply by geekdomotique** [reply]
Hello Ylianst, 
My setup was good, it was the stormshield firewall we use that was too much agressive to let websocket pass by default.
Thank u for your support.
